src/azul/queues.py

```diff
@@ -146,11 +146,12 @@ def get_queues(cls, queue_names: List[str]) -> Mapping[str, Any]:
         return {queue_name: sqs.get_queue_by_name(QueueName=queue_name) for queue_name in queue_names}
 
     @classmethod
-    def count_messages(cls, queues: Mapping[str, Any]) -> int:
+    def count_messages(cls, queues: Mapping[str, Any], log_total: bool = True) -> int:
```

Comment:
You and your modal args. A function with a complex signature is a bad smell because it indicates that the method doesn't just do one thing or doesn't do it well or is a leaky abstraction.  Logging is an auxiliary concern that shouldn't pollute method signatures unless the purpose of those methods is exclusively logging. 

In this particular case, I don't see the point. Why can't that one call site that's passing False for this argument log the total? If there's a redundant log line so be it or maybe that can be solved more elegantly.

Reply:
Agreed. I got caught up in trying to have `count_messages()` support multiple use cases when I should have been considering if it needed to.